src/DotVVM.Framework/Controls/KnockoutHelper.cs

```diff
@@ -215,23 +215,39 @@ private static string GetPostBackHandlersScript(DotvvmBindableObject control, st
         private static string GenerateHandlerOptions(DotvvmBindableObject handler, Dictionary<string, object> options)
         {
             JsExpression optionsExpr = new JsObjectExpression(
-                options.Where(o => o.Value != null).Select(o => new JsObjectProperty(o.Key,
-                    o.Value is IValueBinding b ?
-                    (JsExpression)new JsIdentifierExpression(
-                        JavascriptTranslator.FormatKnockoutScript(b.GetParametrizedKnockoutExpression(handler, unwrapped: true), new ParametrizedCode("c"), new ParametrizedCode("d"))) :
-                    new JsLiteral(o.Value is IBinding ? ((IStaticValueBinding)o.Value).Evaluate(handler) : o.Value)))
+                options.Where(o => o.Value != null).Select(o => new JsObjectProperty(o.Key, TransformOptionValueToExpression(handler, o.Value)))
             );
+
             if (options.Any(o => o.Value is IValueBinding))
+            {
                 optionsExpr = new JsFunctionExpression(
                     new[] { new JsIdentifier("c"), new JsIdentifier("d") },
                     new JsBlockStatement(new JsReturnStatement(optionsExpr))
                 );
+            }
 
             optionsExpr.FixParenthesis();
             var script = new JsFormattingVisitor().ApplyAction(optionsExpr.AcceptVisitor).GetParameterlessResult();
             return script;
         }
 
+        private static JsExpression TransformOptionValueToExpression(DotvvmBindableObject handler, object optionValue)
+        {
+            switch (optionValue)
+            {
+                case IValueBinding binding:
+                    return new JsIdentifierExpression(
+                        JavascriptTranslator.FormatKnockoutScript(binding.GetParametrizedKnockoutExpression(handler, unwrapped: true),
+                            new ParametrizedCode("c"), new ParametrizedCode("d")));
+                case IStaticValueBinding staticValueBinding:
+                    return new JsLiteral(staticValueBinding.Evaluate(handler));
+                case JsExpression expression:
+                    return expression;
+                default:
```

Comment:
To prevent misconceptions with using command binding in the postback handers, it would be nice to throw an exception that general `IBinding` is not supported.

Reply:
okay, I'll add the exception